pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1827,15 +1827,57 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 		}
 
 		// Set VM CPU features
+		existingFeatures := make(map[string]struct{})
 		if vmi.Spec.Domain.CPU.Features != nil {
 			for _, feature := range vmi.Spec.Domain.CPU.Features {
+				existingFeatures[feature.Name] = struct{}{}
 				domain.Spec.CPU.Features = append(domain.Spec.CPU.Features, api.CPUFeature{
 					Name:   feature.Name,
 					Policy: feature.Policy,
 				})
 			}
 		}
 
+		/*
+						Libvirt validation fails when a CPU model is usable
+						by QEMU but lacks features listed in
+						`/usr/share/libvirt/cpu_map/[CPU Model].xml` on a node
+
+						To avoid the validation error mentioned above we will disable
+						some of the features in the `/usr/share/libvirt/cpu_map/[CPU Model].xml` files
+						for some of the cpu models.
+						Examples of validation error:
+			    		https://bugzilla.redhat.com/show_bug.cgi?id=2122283
+						https://gitlab.com/libvirt/libvirt/-/issues/304
+
+						This is temporary solution pending a permanent fix from Libvirt.
```

Comment:
Would you add a URL to an open libvirt request? KubeVirt should not be aware of `mpx` and which models don't really require it.

Reply:
Regardless of other layered products that Kubevirt is relying on, KubeVirt should be able to "defend" itself and allow the administrator to control which flags should be disabled globally in the cluster. 
Other layered products are taking a similar approach: https://docs.openstack.org/nova/2023.2/configuration/config.html#libvirt.cpu_model_extra_flags

On the other hand, the currently proposed approach should change, from my pov, and move the decision to the administrator's hands.

Reply:
> On the other hand, the currently proposed approach should change, from my pov, and move the decision to the administrator's hands.

Why must we put additional burden on the cluster-admin? Most of them (me included!) don't know and don't like to know what `mpx` is. I don't know how many Nova operators have the tools to make such a decision. If possible, I would postpone exposing a new KubeVirt configurable for this until a clear request comes our way.

Reply:
@dankenigsberg This is not directly related to `mpx` at all. This is to allow KubeVirt to be able to react and mitigate issues with CPU models quickly instead of waiting for patches from underline projects (libvirt and QEMU). Even now we still have an issue `svm`. `mpx` is just another case.
When CPU vulnerabilities arise it's possible to mitigate these on a cluster level quickly (Specter / Meltdown)

Reply:
discussion continued on https://github.com/kubevirt/kubevirt/pull/11407